codegen/service_test.go

```diff
@@ -61,7 +61,13 @@ func TestExceptionValidation(t *testing.T) {
 	}
 	assert.NoError(t, err)
 	method.ExceptionsIndex = map[string]codegen.ExceptionSpec{"test": {}}
-	err = m.SetDownstream("Bar", "argNotStruct", clientSpec, "argNotStruct", nil, nil, nil, pkgHelper)
+	e := &codegen.EndpointSpec{
+		ThriftServiceName: "Bar",
+		ThriftMethodName:  "argNotStruct",
+		ClientSpec:        clientSpec,
+		ClientMethod:      "argNotStruct",
```

Comment:
The fact that you only need to specify these fields to run the tests says you don't really need an endpointSpec.

Reply:
this test took a shortcut to cover exception validation case 

-> `	method.ExceptionsIndex = map[string]codegen.ExceptionSpec{"test": {}}`

look at how it construct all the service, method, clientSpec required for a test, pretty sad

if we still have backend/ , where a place for bad, un-generate-able thrift to live, we can write a better integration test with all its required component resolved by zanzibar modules, and we can just have 

`m.SetDownstream(e, pkgHelper)` here instead of `m.SetDownstream(<10 different positional params>)`